### PR TITLE
EVG-13075 Failed Downstream patches should show a failed badge for the upstream patch

### PR DESCRIPTION
**Pull request by malikchaya2**
[EVG-13075](https://jira.mongodb.org/browse/EVG-13075)

### Description 
A patch that has child patches should display a status that reflects the status of the child patches. 

### Testing 
Tested using the graphql playground with the following query: 
`query Patch {
    patch(
      id: "60ec9d6d9dbe322c63945089"
    ) {
    id
    status
    childPatches{
      id
      status
    }
  }
}`

### CODE REVIEW

File: graphql/resolvers.go [hunk 1]
```diff
@@ -971,8 +971,25 @@ func (r *queryResolver) Patch(ctx context.Context, id string) (*restModel.APIPat
...
+		// determine what the patch status should be given the status of
+		// the child patches
+		sort.Sort(restModel.PatchesByStatus(allPatches))
+		// after sorting, the first patch will be the one with the highest priority
+		patch.Status = allPatches[0].Status
+	}
+
 	// If theres an aborted task we should set the patch status to aborted if there are no other failures
 	if utility.StringSliceContains(statuses, evergreen.TaskAborted) {
 		if len(utility.StringSliceIntersection(statuses, evergreen.TaskFailureStatuses)) == 0 {
```
Comment (khelif96) [inline graphql/resolvers.go:995]: We should create a separate status resolver for the patch object and move all this logic there. I started doing this in a pr i am working on but that is still not ready.

Comment (malikchaya2) [reply graphql/resolvers.go]: discussed offline

**Review by khelif96 — COMMENTED** [review]
Other comment about if we need to sort is discussed offline.

**Review by khelif96 — APPROVED** [review]

File: graphql/resolvers.go [hunk 2]
```diff
@@ -971,6 +971,25 @@ func (r *queryResolver) Patch(ctx context.Context, id string) (*restModel.APIPat
...
 		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for patch :%s ", err.Error()))
 	}
+
+	if len(patch.ChildPatches) > 0 {
+		allPatches := []restModel.APIPatch{}
+		allPatches = append(allPatches, *patch)
+		for _, cp := range patch.ChildPatches {
+			allPatches = append(allPatches, cp)
+			// add the child patch tasks to tasks so that we can consider their status
+			childPatchTasks, _, err := r.sc.FindTasksByVersion(*cp.Id, opts)
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply graphql/resolvers.go:981]: discussed offline, opts already passed in

**Pull request merged**